src/toil/cwl/cwltoil.py

```diff
@@ -1657,7 +1657,10 @@ def run(self, file_store: AbstractFileStore):
                                 if isinstance(
                                     promises[s], (CWLJobWrapper, CWLGather)
                                 ) and not promises[s].hasFollowOn(wfjob):
-                                    promises[s].addFollowOn(wfjob)
+                                    try:
+                                        promises[s].addFollowOn(wfjob)
+                                    except ConflictingPredecessorError:
+                                        pass
```

Comment:
How do we know this error is safe to ignore here?

Reply:
I added a comment and changed the logic to a check (though the implementations are equivalent).  It was being added prior and I could change it there, but I think that this is simplest.